File: backend/app/Models/Issue.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Issue extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'project_id',
        'key_number',
        'summary',
        'description',
        'status',
        'priority',
        'issue_type',
        'estimated_hours',
        'actual_hours',
        'assignee_id',
        'creator_id',
        'category',
        'milestone_id',
        'due_date',
        'version',
        'embedding',
    ];

    protected $casts = [
        'estimated_hours' => 'float',
        'actual_hours'    => 'float',
        'due_date'        => 'date:Y-m-d',
    ];

    public function project(): BelongsTo
    {
        return $this->belongsTo(Project::class);
    }

    public function milestone(): BelongsTo
    {
        return $this->belongsTo(Milestone::class);
    }

    /**
     * Get the user that created the issue.
     */
    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'creator_id');
    }

    /**
     * Get the user assigned to the issue.
     */
    public function assignee(): BelongsTo
    {
        return $this->belongsTo(User::class, 'assignee_id');
    }

    /**
     * Get the comments for the issue.
     */
    public function comments(): HasMany
    {
        return $this->hasMany(Comment::class);
    }

    /**
     * Get the history for the issue.
     */
    public function history(): HasMany
    {
        return $this->hasMany(IssueHistory::class);
    }

    /**
     * Get the full issue key (e.g., PROJ-1).
     */
    public function getFullKeyAttribute(): string
    {
        return ($this->project->key ?? 'UNK') . '-' . $this->key_number;
    }
}

File: android/app/src/main/java/com/syncmind/android/data/model/Issue.kt
package com.syncmind.android.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Issue(
    val id: Int,
    val project_id: Int,
    val key: String,
    val key_number: Int,
    val summary: String,
    val description: String? = null,
    val status: String,
    val priority: String,
    val issue_type: String,
    val estimated_hours: Double? = null,
    val actual_hours: Double? = null,
    val assignee_id: Int? = null,
    val milestone_id: Int? = null,
    val due_date: String? = null,
    val creator_id: Int,
    val created_at: String,
    val updated_at: String,
    val comments_count: Int? = null,
    val project_name: String? = null,
    val project_key: String? = null,
    val assignee: User? = null,
    val creator: User? = null,
    val comments: List<IssueComment> = emptyList(),
    val history: List<IssueHistory> = emptyList()
)

@Serializable
data class IssueComment(
    val id: Int,
    val issue_id: Int? = null,
    val user_id: Int? = null,
    val content: String,
    val created_at: String,
    val updated_at: String,
    val user: User? = null
)

@Serializable
data class IssueHistory(
    val id: Int,
    val issue_id: Int? = null,
    val user_id: Int? = null,
    val field: String? = null,
    val old_value: String? = null,
    val new_value: String? = null,
    val created_at: String,
    val updated_at: String? = null,
    val user: User? = null
)

@Serializable
data class CreateIssueCommentRequest(
    val content: String
)

@Serializable
data class SummarizeIssueRequest(
    val force: Boolean = false
)

@Serializable
data class ThreadSummary(
    val summary: String,
    val decisions: List<String> = emptyList(),
    val consensus: String? = null,
    val action_items: List<String> = emptyList()
)

@Serializable
data class ThreadSummaryResponse(
    val data: ThreadSummary,
    val cached: Boolean = false
)

@Serializable
data class SuggestIssueRequest(
    val summary: String
)

@Serializable
data class AssigneeSuggestion(
    val assignee_id: Int,
    val reason: String
)

@Serializable
data class IssueAISuggestion(
    val description: String? = null,
    val issue_type: String? = null,
    val priority: String? = null,
    val estimated_hours: Double? = null,
    val assignee_suggestions: List<AssigneeSuggestion> = emptyList()
)

@Serializable
data class IssueAISuggestionResponse(
    val data: IssueAISuggestion
)

@Serializable
data class SimilarIssue(
    val id: Int,
    val project_id: Int,
    val key: String? = null,
    val full_key: String? = null,
    val key_number: Int,
    val summary: String,
    val description: String? = null,
    val status: String,
    val priority: String,
    val issue_type: String? = null,
    val due_date: String? = null,
    val updated_at: String? = null,
    val comments_count: Int? = null,
    val similarity: Double,
    val project_name: String? = null,
    val project_key: String? = null,
    val assignee_id: Int? = null,
    val assignee: User? = null
)

@Serializable
data class SimilarIssuesResponse(
    val data: List<SimilarIssue>
)

@Serializable
data class IssuesSummary(
    val assigned_to_me: Int,
    val overdue: Int,
    val high_priority: Int,
    val unassigned: Int,
    val project_name: String
)

@Serializable
data class IssuesSummaryResponse(
    val data: IssuesSummary
)

@Serializable
data class IssueCommentResponse(
    val data: IssueComment
)

@Serializable
data class CreateIssueRequest(
    val summary: String,
    val description: String? = null,
    val issue_type: String,
    val priority: String = "normal",
    val status: String = "open",
    val estimated_hours: Double? = null,
    val due_date: String? = null
)

@Serializable
data class UpdateIssueRequest(
    val summary: String? = null,
    val description: String? = null,
    val issue_type: String? = null,
    val priority: String? = null,
    val status: String? = null,
    val estimated_hours: Double? = null,
    val actual_hours: Double? = null,
    val due_date: String? = null
)
